File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if(len(nums) > 1):
            pivot = random.choice(nums)
            left, right, eq = [], [], []
            for num in nums:
                if(num > pivot):
                    right.append(num)
                elif(num < pivot):
                    left.append(num)
                else:
                    eq.append(num)

            return self.sortArray(left) + eq + self.sortArray(right)
        
        return nums
            

File: 0912-sort-an-array/0912-sort-an-array.go
func sortArray(nums []int) []int {
    var upto = 1_00_001
    var arr = make([]int, upto)
    for _, num := range nums {
        arr[num + 50000]++
    }

    var index = 0
    for i, v := range arr {
        for range v {
            nums[index] = i - 50000
            index++
        }
    }
    return nums
}